actioncable/lib/action_cable/engine.rb

```diff
@@ -22,7 +22,9 @@ class Railtie < Rails::Engine # :nodoc:
 
     initializer "action_cable.set_configs" do |app|
       options = app.config.action_cable
-      options.allowed_request_origins ||= "http://localhost:3000" if ::Rails.env.development?
+      if ::Rails.env.development? && !ActionCable.server.config.allowed_request_origins
+        options.allowed_request_origins ||= "http://localhost:3000"
+      end
```

Comment:
A better fix would be to push this into the `on_load` down below and do:

```ruby
config.allowed_request_origins |= options.allowed_request_origins || [ 'http://localhost:3000' ]
```

Also necessitates we default `allowed_request_origins` to an array in the server configuration's initialize method.